guides/configuration.md

```diff
@@ -0,0 +1,102 @@
+
+When you override a view from a Gem, there is the possibility of difficulties when upgrading, because Controller code within the Gem might have changed, such as for example the name of an instance variable. Be careful when upgrading, and always read `CHANGELOG.md`.
+
+### In the backend
+
+We do not recommend overriding the admin backend in the same way, since some of the extensions you might be running (see below) depend on them having the HTML structure present in the version of Solidus they were built for. If you need to add custom functionality to Solidus' backend, we recommend using the `Deface` gem. Find it [here](https://github.com/spree/deface). Overriding backend views with Deface is tricky - be sure to have good tests!
+
+## Changing assets
+
+If you need to change the look and feel of your app without changing the HTML structure, you can customize the CSS, image and JavaScript assets by amending the files in `vendor/assets/*/*/all.js`. They mirror the structure of the `app/assets` folder. Have a look at the [Spree Guide](http://guides.spreecommerce.org/developer/asset.html) for overriding assets, it applies to Solidus as-is.
+
+## Installing extensions
+
+The Solidus ecosystem is rich in extensions that modify Solidus' behaviour. The extensions are distributed as Gems, so it's often as easy as adding the extension to your Gemfile and running an install generator. Sometimes, however, the extension needs more work, such as adding custom configuration in its own initializer. Please refer to the individual extension's Readme file.
+
+We maintain an [extensions compatibility matrix](http://extensions.solidus.io/). This is a good starting point for finding out which extensions are available and whether they work with your chosen Solidus version.
+
+Every extension adds another dependency to your app. Choose wisely.
+
+## Custom service classes
+
+The Solidus configuration object from above has some getters and setters for classes as extension points. After evaluating whether you need to go down this route, take some time to study the standard implementation for the class you want to modify, and subclass or re-implement it. The service class extension points can be found [here](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/app_configuration.rb#L277-L383).
+
+These extension points can alter Solidus' behaviour quite drastically. If you make mistakes when implementing them, your store will break. Good luck!
+
+## Monkey-Patching Solidus
+
+Solidus is setup so that any file in the `app` directory with the suffix `_decorator.rb` will be auto-loaded like any model or controller files.
+
+For example, if you want to add a method to the `Spree::Order` model, you would have a file called `app/models/spree/order_decorator.rb` with the following contents:
+
+```
+module Spree
```

Comment:
It's not necessary to use the `Spree` `module` namespace and can even lead to strange issues. 

Maybe just use none or `module YourStore`